core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -717,4 +729,37 @@ static void prepareResizeIndexSettings(ClusterState currentState, Set<String> ma
             .put(IndexMetaData.INDEX_RESIZE_SOURCE_NAME.getKey(), resizeSourceIndex.getName())
             .put(IndexMetaData.INDEX_RESIZE_SOURCE_UUID.getKey(), resizeSourceIndex.getUUID());
     }
+
+    /**
+     * Returns a default number of routing shards based on the number of shards of the index. The default number of routing shards will
+     * allow any index to be split at least once and at most 10 times by a factor of two. The closer the number or shards gets to 1024
+     * the less default split operations are supported
+     */
+    public static int calculateNumRoutingShards(int numShards, Version indexVersionCreated) {
+        if (indexVersionCreated.onOrAfter(Version.V_7_0_0_alpha1)) {
+            // only select this automatically for indices that are created on or after 7.0
+            int base = 10; // logBase2(1024)
```

Comment:
I think it's worth sharing the 1024 constant with the max value in `buildNumberOfShardsSetting` to be clearer where this comes from?

Reply:
not sure how you envisioned this to work?

Reply:
I was just thinking of having a `public static final int DEFAULT_MAX_NUM_SHARDS = 1024` somewhere, and use it in `buildNumberOfShardsSetting` and here: `int base = Math.log(DEFAULT_MAX_NUM_SHARDS)/Math.log(2)`